### PR TITLE
Pull request fixes #16

### PR DESCRIPTION
**Pull request by RedStylzZ**

### CODE REVIEW

File: backend/src/main/java/com/github/redstylzz/backend/controller/LoginController.java [hunk 1]
```diff
@@ -20,11 +19,16 @@ public LoginController(LoginService service) {
         this.service = service;
     }
 
+    @GetMapping
+    public boolean checkLogin(HttpServletRequest request) {
+        return service.checkLogin(request);
+    }
+
```
Comment (Flooooooooooorian) [inline backend/src/main/java/com/github/redstylzz/backend/controller/LoginController.java:26]: What is this for?

File: frontend/src/components/Categories.tsx [hunk 2]
```diff
@@ -2,9 +2,14 @@ import CategoryItem from "./CategoryItem";
...
-    const {categories, config, deleteCategory} = props;
-    if (!categories || !Array.isArray(categories)) return null;
+interface CategoriesProps {
+    categories: Category[],
+    config: ITokenConfig,
+    deleteCategory: IDeleteCategory
+}
+
+export default function Categories({ categories, config, deleteCategory }: CategoriesProps) {
+    if (!Array.isArray(categories)) return null;
```
Comment (Flooooooooooorian) [inline frontend/src/components/Categories.tsx:12]: You dont need this either.
```suggestion
```

File: frontend/src/components/Payments.tsx [hunk 3]
```diff
@@ -1,9 +1,12 @@
...
 
-export default function Payments(props: { payments: Payment[] }) {
-    const {payments} = props
-    if (!payments || !Array.isArray(payments)) return null;
+interface PaymentsProps {
+    payments: Payment[]
+}
+
+export default function Payments({ payments }: PaymentsProps) {
+    if (!Array.isArray(payments)) return null;
```
Comment (Flooooooooooorian) [inline frontend/src/components/Payments.tsx:9]:
```suggestion
```

File: frontend/src/components/RequireAuth.tsx [hunk 4]
```diff
@@ -1,9 +1,13 @@
...
 import {AuthContext} from "../context/AuthProvider";
 import {Navigate} from "react-router-dom";
+import {ILoginController} from "../models/ControllerTypes";
+import LoginController from "../controllers/LoginController";
 
 export default function RequireAuth({children}: { children: ReactElement<any, any> }) {
-    const {token} = useContext(AuthContext)
+    const {token, logout} = useContext(AuthContext)
+    const controller: ILoginController = LoginController()
+    controller.checkLoggedIn().catch( () => logout());
```
Comment (Flooooooooooorian) [inline frontend/src/components/RequireAuth.tsx:10]: I would not ask the backend  if your still logged in. 
I does not make anything more secure,
You are making a lot of request to your backend,
every time changing the route the user must wait for the request to finish,

You could check the expireationTime of the token locally

**Review by pakantu — APPROVED** [review]

File: frontend/src/pages/CategoryPage.tsx [hunk 5]
```diff
@@ -23,29 +20,32 @@ export default function CategoryPage() {
...
-        const categoryName = formElements.categoryInput.value.trim()
+        const categoryName: string = categoryInput.trim()
 
         if (categoryName && categoryName.length > 0) {
             controller.addCategory(categoryName).then(setCategories)
         }
-        formElements.categoryInput.value = "";
+
+        // @ts-ignore
+        event.currentTarget.elements[0].value = ""
```
Comment (Flooooooooooorian) [inline frontend/src/pages/CategoryPage.tsx:30]: Add the value prop to the input. Then the input is fully controlled and you can reset it via the state... :)

**Review by Flooooooooooorian — APPROVED** [review]

**Pull request merged**